source/agora/network/Clock.d

```diff
@@ -156,3 +158,21 @@ public class Clock
             this.net_time_offset = time_offset;
     }
 }
+
+public class MockClock : Clock
```

Comment:
Mocking stuff should be moved to one of the test modules like [Base.d](https://github.com/bpfkorea/agora/blob/v0.x.x/source/agora/test/Base.d).

There is already a `TestClock` there for example: https://github.com/bpfkorea/agora/blob/30956d13635a1e1ec77b7ce7c5f157f8d851c67d/source/agora/test/Base.d#L1441

Reply:
I saw TestClock, but I don't feel it is easy enough to use in unit tests: it requires creating delegates and passing task manager... Even if I created a new constructor that only sets the time there are inherited methods like 'synchronize' that would use the delegate. Also I think it would have been better to define the Clock class as an interface.

Reply:
Sure! I meant it would be good to place it either there or in the module where the `MockClock` is used.

Reply:
unfortunately putting MockClock into Base cases:
object.Error@rt/minfo.d(371): Cyclic dependency between module constructors/destructors of agora.node.Ledger and agora.node.Validator
agora.node.Ledger* ->
agora.test.Base ->
agora.node.Validator* ->
agora.node.Ledger*

when I am trying to unit test agora.node.Ledger

Reply:
Damn. D strikes again. Just leave it there then.